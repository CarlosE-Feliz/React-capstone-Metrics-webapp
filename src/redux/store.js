import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import coinsReducer from './coinsAPI';

const reducers = combineReducers({
  coins: coinsReducer,
});

const store = createStore(
  reducers,
  applyMiddleware(logger, thunk),
);

export default store;
