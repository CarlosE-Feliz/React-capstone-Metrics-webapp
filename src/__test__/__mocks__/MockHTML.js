import { useSelector, useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Card } from 'semantic-ui-react';
import { getOneCoin } from '../../redux/coinsAPI';

const Data = ({ id, name, symbol }) => {
  const post = useSelector((state) => state.rock);
  const dispatch = useDispatch();
  const oneCoin = () => {
    dispatch(getOneCoin(post, id));
  };

  return (
    <NavLink
      style={{ textDecoration: 'none' }}
      className="propss"
      key={name}
      name={name}
      id={id}
      to="/Details"
    >
      <div key={id} id={id}>
        <Card.Content>
          <Card.Header
            id={id}
            className="cardprops"
            onClick={oneCoin}
          >
            {symbol}
          </Card.Header>
          <Card.Description
            className="cardprops"
          >
            {name}
          </Card.Description>
        </Card.Content>
      </div>
    </NavLink>
  );
};

Data.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
};

export default Data;
