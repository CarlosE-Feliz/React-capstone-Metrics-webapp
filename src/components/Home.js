import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { Card } from 'semantic-ui-react';
import { getOneCoin } from '../redux/coinsAPI';
import '../styles/home.css';

const Home = () => {
  const data = useSelector((state) => state.coins);
  const dispatch = useDispatch();
  const oneCoin = (e) => {
    dispatch(getOneCoin(data, e.target.id));
  };

  return (
    <div
      className="grid"
      role="button"
      tabIndex={0}
      key="main"
      onClick={oneCoin}
      onKeyDown={oneCoin}
    >
      {
          data.map((item) => (
            <NavLink
              style={{ textDecoration: 'none' }}
              className="items"
              key={item.name}
              name={item.name}
              id={item.id}
              to="/Details"
            >
              <div key={item.id} id={item.id}>
                <Card.Content id={item.id}>
                  <Card.Header
                    id={item.id}
                    className="cardItem"
                  >
                    <h2 id={item.id}>
                      {item.symbol}
                    </h2>
                  </Card.Header>
                  <Card.Description
                    id={item.id}
                    className="cardItem"
                  >
                    <h6 id={item.id}>
                      {item.name}
                    </h6>
                  </Card.Description>
                </Card.Content>
              </div>
            </NavLink>
          ))
      }
    </div>
  );
};
export default Home;
