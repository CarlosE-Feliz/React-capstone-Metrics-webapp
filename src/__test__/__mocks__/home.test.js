/* eslint-disable no-unused-vars */
import { render, fireEvent, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import Home from '../../components/Home';
import { Data } from './MockHTML';

describe('Test button Homes', () => {
  it('snapshot test', async () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home>
            <Data id="1" name="test" symbol="TEST" />
          </Home>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('test click', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Data id="1" name="test" symbol="BTC" />
        </BrowserRouter>
      </Provider>,
    );
    screen.getByText('test');
    expect(screen.getByText('BTC')).toMatchSnapshot();
  });
});
