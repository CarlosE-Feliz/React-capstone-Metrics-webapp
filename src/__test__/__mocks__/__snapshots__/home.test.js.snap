// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button Homes snapshot test 1`] = `
<div
  className="grid"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
/>
`;

exports[`Test button Homes test click 1`] = `
<div
  class="header cardprops"
  id="1"
>
  BTC
</div>
`;
